core/src/main/java/dk/alexandra/fresco/lib/crypto/mimc/MimcConstants.java

```diff
@@ -0,0 +1,50 @@
+package dk.alexandra.fresco.lib.crypto.mimc;
+
+import dk.alexandra.fresco.framework.util.Pair;
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+/**
+ * Deterministically seeded generator for MiMC round constants.
+ */
+public class MimcConstants {
+
+  /**
+   * Map used to cache mimc round constants.
+   */
+  private static Map<Pair<Integer, BigInteger>, BigInteger> roundConstants = new HashMap<>();
+
+  private MimcConstants() {
+    // Should not be instantiated
+  }
+
+  /**
+   * Returns a random but deterministic constant depending on the round and the field size. It might
+   * be a better idea to pre-generate the constants using SecureRandom, and distribute them among
+   * the participants if run in production.
```

Comment:
This javadoc leads me to believe these constants should not be generated on the fly but rather be supplied to the mimc de-/encryption?

Reply:
That is a really good point. I wonder if it was done like this originally because it is dependent on the modulus, which is generally not known at construction.

Reply:
Actually, I am not really sure what would be a good way to generate and represent these constants.